service/dynamodb/dynamodbattribute/decode.go

```diff
@@ -185,14 +185,32 @@ func (d *Decoder) decodeBinary(b []byte, v reflect.Value) error {
 		return nil
 	}
 
-	switch v.Interface().(type) {
-	case []byte:
+	if v.Kind() != reflect.Slice {
+		return &UnmarshalTypeError{Value: "binary", Type: v.Type()}
+	}
+
+	if v.Type() == byteSliceType {
+		// Optimization for []byte types
 		if v.IsNil() || v.Cap() < len(b) {
 			v.Set(reflect.MakeSlice(byteSliceType, len(b), len(b)))
 		} else if v.Len() != len(b) {
 			v.SetLen(len(b))
 		}
 		copy(v.Interface().([]byte), b)
+		return nil
+	}
+
+	switch v.Type().Elem().Kind() {
+	case reflect.Uint8:
+		// Fallback to reflection copy for type aliased of []byte type
+		if v.IsNil() || v.Cap() < len(b) {
```

Comment:
Not sure, But with that said not sure it would even need to because a slice has a `len` internal field on its data structure. So it won't need to calculate value each call.